lib/fuel-rest.js

```diff
@@ -84,16 +84,19 @@ class FuelRest {
 				return tokenInfo;
 			})
 			.then(tokenInfo => {
-				let retry = false;
+				let retry = options.retry || false;
 				const consolidatedOpts = {};
 				const authOptions = clone(options.auth);
+				const headers = options.headers;
```

Comment:
It doesn't work because by default this `headers` here is `undefined`.

It's better to update the unit tests to make sure the token actually changed.

I tried to update this test `"should try request again if 401 stating token is invalid"` on your branch:

```
var firstAttempt = true;
sinon.stub(FuelAuth.prototype, "getAccessToken").callsFake(() => {
			var accessToken = firstAttempt ? "testing" : "retry";
			firstAttempt = false;
			return new Promise(resolve =>
				resolve({ accessToken, expiresIn: 3600 })
			);
		});
```

Then checked the first parameter of `_makeRequest`:
```
_makeRequest(consolidatedOpts, callback) {
		console.log("consolidatedOpts", consolidatedOpts);
```

But the results are:

```
consolidatedOpts { req:
   { method: 'GET',
     uri: 'http://127.0.0.1:4550/invalid/token',
     headers:
      { 'User-Agent': 'node-fuel/3.0.3',
        'Content-Type': 'application/json',
        Authorization: 'Bearer testing' } },
  auth: { force: true },
  accessToken: 'testing',
  retry: true }
consolidatedOpts { req:
   { method: 'GET',
     uri: 'http://127.0.0.1:4550/invalid/token',
     headers:
      { 'User-Agent': 'node-fuel/3.0.3',
        'Content-Type': 'application/json',
        Authorization: 'Bearer testing' } },
  auth: { force: true },
  accessToken: 'retry',
  retry: false }
```

You can see the actual Authorization header doesn't change on retry.